python/cuml/cuml/ensemble/randomforest_common.pyx

```diff
@@ -72,6 +72,31 @@ class BaseRandomForestModel(UniversalBase):
 
     classes_ = CumlArrayDescriptor()
 
+    @classmethod
+    def _criterion_to_split_criterion(cls, criterion):
+        """Translate sklearn-style criterion string to cuML equivalent"""
+        if criterion is None:
+            split_criterion = cls._default_split_criterion
+        else:
+            if criterion == "squared_error":
+                split_criterion = "mse"
+            elif criterion == "absolute_error":
+                split_criterion = "mae"
+            elif criterion == "poisson":
+                split_criterion = "poisson"
+            elif criterion == "gini":
+                split_criterion = "gini"
+            elif criterion == "entropy":
+                split_criterion = "entropy"
```

Comment:
Doing this with a mapping and `try`/`except` (or `get` and branch on `None`) would be a bit more idiomatic.

```python
try:
    split_criterion = criterions[criterion]
except KeyError:
    raise NotImplementedError(...) from None
```

Not a big deal though.

Reply:
I thought about this, but we may want to do something a little more sophisticated with this translation in the future (taking into account the value of two separate hyperparameters) to provide more robust support for cuML CPU. Right now, this PR punts on cuML CPU translation, but that should be fixed later on. I _think_ this structure is slightly better for handling that more complex logic, but we can always simplify when we go to actually do the work.